src/openingHours.ts

```diff
@@ -104,6 +115,150 @@ export class OpeningHours {
+          return formatHours[i].hour;
+        }
+      }
+    }
+    return null;
+  };
+
+  private openUntil = (formatHours: FormatHours[], date: Date): string => {
+    if (formatHours.length > 1) {
+      const now = this.constructDateFromTime(`${date.getHours()}:${date.getMinutes()}`);
+      for (let i = 1; i < formatHours.length; i = i + 2) {
+        if (now < formatHours[i].date || now > formatHours[formatHours.length - 1].date) {
+          return formatHours[i].hour;
+        }
+      }
+    }
+    return `Aujourd'hui`;
+  };
+
+  public checkOpeningTime = (date: Date) => {
+    const isOpenNow = this.isOpenOn(date);
+
+    const allHours = this.getTable();
+
+    const dayIndex = date.getDay();
+    const daySimpleIndex = SIMPLE_DAYS_NAMES[dayIndex];
+    const todayHours = allHours[daySimpleIndex];
+    if (!todayHours || todayHours.length === 0) {
+      return this.nextDayOpeningTime(allHours, date);
+    }
+    const formatHours: FormatHours[] = [];
+    todayHours.map(todayHour =>
+      todayHour
+        .split('-')
+        .map(hour => formatHours.push({ date: this.constructDateFromTime(hour), hour })),
+    );
+    if (isOpenNow) {
+      return {
+        open: true,
+        openUntil: this.openUntil(formatHours, date),
+        openAt: null,
+        weekDay: null,
+      };
+    }
+    if (!isOpenNow) {
+      const hour = this.openAt(formatHours, date, undefined);
+
+      if (!hour) {
+        return this.nextDayOpeningTime(allHours, date);
+      }
+      return {
+        open: false,
+        openUntil: null,
+        openAt: hour,
+        weekDay: null,
+      };
+    }
+  };
+
+  private nextDayOpeningTime = (allHours: OpeningHoursData, date: Date) => {
+    const nextDay: NextOpeningDay = this.getNextOpeningDay(date);
+    const nextDayHours = allHours[nextDay.nextDay];
+    if (!nextDayHours || nextDayHours.length === 0) {
+      return null;
+    }
+    const formatNextDayHours: FormatHours[] = [];
+    nextDayHours.map(nextDayHour =>
+      nextDayHour
+        .split('-')
+        .map(hour => formatNextDayHours.push({ date: this.constructDateFromTime(hour), hour })),
+    );
+    const hour = this.openAt(formatNextDayHours, date, nextDay.nextDay);
+    return {
+      open: false,
+      openUntil: null,
+      openAt: hour,
+      weekDay: nextDay.isTomorrow ? 'tomorrow' : this.formatWeekDay(nextDay.nextDay) ?? null,
+    };
+  };
+
+  public checkOpeningTimeNow = () => {
+    return this.checkOpeningTime(new Date());
+  };
+
+  public formatWeekDay = (weekDay: string): string => {
+    switch (weekDay) {
+      case 'mo':
+        return 'monday';
+
+      case 'tu':
+        return 'tuesday';
+
+      case 'we':
+        return 'wednesday';
+
+      case 'th':
+        return 'thirsday';
+
+      case 'fr':
+        return 'friday';
+
+      case 'sa':
+        return 'saturday';
+
+      case 'su':
+        return 'sunday';
+
+      default:
+        return 'Tomorrow';
+    }
+  };
+
```

Comment:
In my opinion we don't need the full name. Just the day index is enough.

```
const DAY_INDEX {
  'su': 0,
  'mo': 1,
...
}
```

So thiw way we don't handle at all internationalization.